functional_programming/lists/exercise2.r

```diff
@@ -18,3 +18,6 @@ x$f(x$z)       # 2
 f(x$z)         # x$z
 x$f(z)         # Error in x$f(z) : object 'z' not found
 f(z)           # Error: could not find function "f"
+
+## Igor
+# e. It depends on whether f, z are functions or data
```

Comment:
I think you're free to assume that `f` is a function, given that it is treated as such in `with(x, f(z))`.  Similarly, `x` must be a list, environment, or data.frame given `with`, and can furthermore be assumed to be a list or environment given `$` accessing in the answers.  What `z` is doesn't seem to matter.